ansible/roles/preflights/tasks/main.yaml

```diff
@@ -0,0 +1,393 @@
+
+  - name: check required OS package registries are accessible for rpm (rhel8 only)
+    uri:
+      url: "{{item}}"
+      status_code: 200
+    with_items:
+      - "{{kubernetes_rpm_repository_url}}/repodata/repomd.xml"
+      - "{{kubernetes_rpm_gpg_key_url}}"
+      - "{{docker_rpm_repository_url}}/repodata/repomd.xml"
+      - "{{docker_rpm_gpg_key_url}}"
+      - "{{libnvidia_container_rpm_repository_package_url}}/repodata/repomd.xml"
+      - "{{libnvidia_container_gpg_key_url}}"
+      - "{{nvidia_container_runtime_rpm_repository_package_url}}/repodata/repomd.xml"
+      - "{{nvidia_container_runtime_gpg_key_url}}"
+    register: result
+    until: result is success
+    retries: 5
+    delay: 3
+    when:
+      - package_versions.enable_repository_installation
+      - ansible_os_family == 'RedHat'
+      - ansible_distribution_major_version|int == 8
+      - rpms_tar_file == ''
+
+  - name: check required OS package registries are accessible for deb
+    uri:
+      url: "{{item}}"
+      status_code: 200
+    with_items:
+      - "{{kubernetes_deb_repository_url}}"
+      - "{{kubernetes_deb_gpg_key_url}}"
+      - "{{docker_deb_repository_url}}/dists/{{docker_deb_release_name}}/{{ docker_deb_component }}/binary-amd64/Packages"
+      - "{{docker_deb_gpg_key_url}}"
+      - "{{libnvidia_container_deb_repository_url}}/repo.conf"
+      - "{{libnvidia_container_gpg_key_url}}"
+      - "{{nvidia_container_runtime_deb_package_url}}/repo.conf"
+      - "{{nvidia_container_runtime_gpg_key_url}}"
+    register: result
+    until: result is success
+    retries: 5
+    delay: 3
+    when:
+      - package_versions.enable_repository_installation
+      - ansible_os_family == 'Debian'
+      - debs_tar_file == ''
+
+  - name: check required OS package registries are accessible for zypper
+    uri:
+      url: "{{item}}"
+      status_code: 200
+    with_items:
+      - "{{kubernetes_zypper_repository_url}}/repodata/repomd.xml"
+      - "{{kubernetes_zypper_gpg_key_url}}"
+      - "{{docker_zypper_repository_url}}/repodata/repomd.xml"
+      - "{{docker_zypper_gpg_key_url}}"
+    register: result
+    until: result is success
+    retries: 5
+    delay: 3
+    when:
+      - package_versions.enable_repository_installation
+      - ansible_os_family == 'Suse'
+
+  # kubernetes checks /proc/swaps lines > 1
+  - name: check memory swap is disabled
+    command: cat /proc/swaps
+    register: memory_swaps
+    failed_when:
+      - memory_swaps is defined
+      - memory_swaps.rc is defined
+      - (memory_swaps.rc != 0 or (memory_swaps.stdout_lines is defined and memory_swaps.stdout_lines|length > 1))
+    changed_when: False
+    when: spec.kubernetes.preflightChecks is not defined or
+          spec.kubernetes.preflightChecks.errorsToIgnore is not defined or
+          not (
+            'all' in spec.kubernetes.preflightChecks.errorsToIgnore or
+            'swap' in spec.kubernetes.preflightChecks.errorsToIgnore
+          )
```

Comment:
There is no `spec.kubernetes.preflightChecks.errorsToIgnore`

Reply:
right I want to pass this as a flag, bear with me